File: tallninja/ReverseInteger.cs
using System;

class Program
{
    static void Main()
    {
        // prompts the user to enter an integer
        Console.WriteLine("Enter an integer:");

        // read the input an try to parse it as integer
        if (int.TryParse(Console.ReadLine(), out int number))
        {
            // call the ReverseDigits method and output the result
            int reversedNumber = ReverseDigits(number);
            Console.WriteLine($"The reversed number is: {reversedNumber}");
        }
        else
        {
            Console.WriteLine("Invalid input. Please enter a valid integer.");
        }
    }

    static int ReverseDigits(int number)
    {
        // check is a number is negative
        bool isNegative = number < 0;

        // convert the number to its absolute value
        number = Math.Abs(number);

        // initialize the reversed number to 0
        int reversedNumber = 0;

        // extract each digit from the number and build the reversed number
        while (number > 0)
        {
            // get the last digit from the number.
            // for example id number is 46 then 46 % 10 is 6
            int digit = number % 10;
            // start assembling the reversed number
            reversedNumber = reversedNumber * 10 + digit;
            // remove the last digit of the number 
            number /= 10;
        }

        // retain the sign of the number
        return isNegative ? -reversedNumber : reversedNumber;
    }
}

File: douglasgatimu/reverse_integer.py
def reverse_integer(int_num):
    is_negative = int_num < 0
    
    if is_negative:
        result = int(str(abs(int_num))[::-1]) * -1  
    else:
        result = int(str(int_num)[::-1])  
    return result

    
if __name__ == "__main__":
    print(f"For input 500, result is {reverse_integer(500)}.") #should return 5. 
    print(f"For input -56, result is {reverse_integer(-56)}.") #should return -65. 
    print(f"For input -90, result is {reverse_integer(-90)}.") #should return -9. 
    print(f"For input 91, result is {reverse_integer(91)}.") #should return 19. 
